weather_server/show_weather/views.py

```diff
@@ -75,4 +76,22 @@ def csv_observation_request(request, station, start, end):
         )
     )
     response.seek(0)
-    return HttpResponse(response)
+    return HttpResponse(response, content_type="text/html")
+
+def png_observation_request(request, station, start, end):
+    """Return PNG of observations in requested time range."""
+    obs = Observation.objects.filter(station__station_id=station,
+                                     obs_date__gte=start,
+                                     obs_date__lte=end)
+
+    #TODO create n sub plots for all relevant parameter or merge them in 1 plot	
+    # TODO how to loop over all elements and extract name along the way?
```

Comment:
Might be easiest to send out a HTTP request and get the CSV from the server. If you read the response into a ``StringIO`` buffer you could pass it to ``pandas`` using ``read_csv`` and you'd have the data as a ``DataFrame`` without having to manipulate the observations directly. The HTTP request shouldn't create a bottleneck because it would all run locally.